adminpage/views.py

```diff
@@ -1,3 +1,272 @@
 from django.shortcuts import render
-
+from codex.baseerror import *
+from codex.baseview import APIView
+from django.contrib.auth.models import User
+from django.contrib.auth import authenticate, login, logout
+from wechat.models import Activity
+from wechat.models import Ticket
+from wechat.views import CustomWeChatView
+from WeChatTicket.settings import SITE_DOMAIN
+import time
+from datetime import datetime, timedelta
 # Create your views here.
+
+
+class Login(APIView):
+
+    def get(self):
+        if not self.request.user.is_authenticated():
+            raise ValidateError('User not logged in')
+        return 0
+
+    def post(self):
+        self.check_input('username', 'password')
+        username = self.input['username']
+        password = self.input['password']
+        user = authenticate(username=username, password=password)
+
+        if not username:
```

Comment:
not necessary judgement!
username and password are not empty, which has been checked in the frontend